Makefile

```diff
@@ -1,96 +1,67 @@
-OUTPUT := .output
-TARGET := event_monitor_ebpf
-KERN_RELEASE := $(shell uname -r)
-KERN_SRC := /lib/modules/$(KERN_RELEASE)/build
-ARCH := $(shell uname -m | sed 's/x86_64/x86/')
+.PHONY: all
+all: build
+
+OUT_DIR := .output
+
+$(OUT_DIR):
+	mkdir -p $@
+
+ARCH := $(shell uname -m)
+bpf_src := tracee/event_monitor_ebpf.c 
+bpf_out := $(OUT_DIR)/event_monitor_ebpf.o
+go_src := $(shell find . -type f -name '*.go' ! -name '*_test.go')
+bin_out := $(OUT_DIR)/tracee
+
+.PHONY: build
+build: $(bin_out)
+
+$(bin_out): $(bpf_out) $(go_src) | $(OUT_DIR)
+	GOOS=linux GOARCH=$(ARCH:x86_64=amd64) go build -v -o $(bin_out)
+
+# .PHONY: build-docker
+# build-docker: clean
+# 	img=$$(docker build --target builder -q  .) && \
+# 	cnt=$$(docker create $$img) && \
+# 	docker cp $$cnt:/tracee/dist - | tar -xf - ; \
+# 	docker rm $$cnt ; docker rmi $$img
+
 LLC ?= llc
 CLANG ?= clang
 LLVM_STRIP ?= llvm-strip
-BPF_C = tracee/${TARGET:=.c}
-BPF_OBJ = ${OUTPUT}/${TARGET:=.o}
-LIBBPF_SRC = $(abspath 3rdparty/libbpf/src)
-LIBBPF_OBJ = $(abspath $(OUTPUT)/libbpf.a)
-CFLAGS ?= -I$(OUTPUT)/usr/include/
-
-SRC = $(shell find . -type f -name '*.go' ! -name '*_test.go' )
-ebpfProgramBase64 = $(shell base64 -w 0 tracee/event_monitor_ebpf.c)
+bpf_compile_tools = $(LLC) $(CLANG) $(LLVM_STRIP)
 
-.PHONY: build
-build: llvm-check $(BPF_OBJ) dist/tracee
+.PHONY: $(bpf_compile_tools) 
+$(bpf_compile_tools): % : check_%
 
-.PHONY: build-docker
-build-docker: clean
-	img=$$(docker build --target builder -q  .) && \
-	cnt=$$(docker create $$img) && \
-	docker cp $$cnt:/tracee/dist - | tar -xf - ; \
-	docker rm $$cnt ; docker rmi $$img
+libbpf_src := 3rdparty/libbpf/src
+libbpf_out_headers_dir := $(OUT_DIR)/usr/include
 
-dist/tracee: $(SRC) tracee/event_monitor_ebpf.c
-	GOOS=linux go build -v -o dist/tracee -ldflags "-X github.com/aquasecurity/tracee/tracee.ebpfProgramBase64Injected=$(ebpfProgramBase64)"
+$(libbpf_src):
+	test -d $(libbpf_src) || ( echo "missing libbpf source, try git submodule update --init" ; false )
 
-.PHONY: test
-test:
-	go test -v ./...
+$(libbpf_out_headers_dir) : | $(OUT_DIR) $(bpf_compile_tools) $(libbpf_src)
```

Comment:
I agree it sounds weird, i'll rethink it. just to explain: initially I called it `libbpf_out` but as a preperation for a future commit where I compile the libbpf `a` file into the `OUT_DIR` I added `headers` to explain. the `dir` is there because this is used as a flag to clang, and there is a different flag for dirs vs files, so it will be clear.